spring-data-mongodb/src/test/java/org/springframework/data/mongodb/repository/query/StringBasedMongoQueryUnitTests.java

```diff
@@ -293,10 +303,36 @@ public void shouldSupportExpressionsInCustomQueries() {
 		assertThat(query.getQueryObject()).isEqualTo(reference.getQueryObject());
 	}
 
+	@Test // GH-3050
+	public void shouldSupportExpressionsAndPropertiesInCustomQueries() {
+
+		ConvertingParameterAccessor accessor = StubParameterAccessor.getAccessor(converter, "Matthews");
+		StringBasedMongoQuery mongoQuery = createQueryForMethod("findByQueryWithExpressionAndProperty", String.class);
+
+		org.springframework.data.mongodb.core.query.Query query = mongoQuery.createQuery(accessor);
+		org.springframework.data.mongodb.core.query.Query reference = new BasicQuery(
+				"{'lastname' : 'Matthews', 'firstname' : 'some-default'}");
+
+		assertThat(query.getQueryObject()).isEqualTo(reference.getQueryObject());
+	}
+
+	@Test // GH-3050
+	public void shouldSupportPropertiesInCustomQueries() {
+
+		properties.put("foo", "bar");
+		ConvertingParameterAccessor accessor = StubParameterAccessor.getAccessor(converter);
+		StringBasedMongoQuery mongoQuery = createQueryForMethod("findByQueryWithProperty");
+
+		org.springframework.data.mongodb.core.query.Query query = mongoQuery.createQuery(accessor);
+		org.springframework.data.mongodb.core.query.Query reference = new BasicQuery("{'lastname' : 'bar'}");
+
+		assertThat(query.getQueryObject()).isEqualTo(reference.getQueryObject());
```

Comment:
what do we expect when the property is missing and does not have a default?

Reply:
IMO it should break (and it does right now, I'll add a test for that)